general_autoencoder.py

```diff
@@ -292,23 +365,30 @@ def fill_feed_dict(self, data, regularization_weighting, ages=None, idxs=None):
                 indexed_ages = ages[idxs]
             else:
                 indexed_ages = ages
+                
+            if age_adjusted_data is not None:
+                indexed_age_adjusted_data = age_adjusted_data[idxs, :]
+            else:
+                indexed_age_adjusted_data = age_adjusted_data
             indexed_data = data[idxs, :]
         else:
             indexed_ages = ages
             indexed_data = data
+            indexed_age_adjusted_data = age_adjusted_data
 
        
         if self.need_ages:
             feed_dict = {
                 self.ages:indexed_ages, 
                 self.X:indexed_data, 
+                self.age_adjusted_X:indexed_age_adjusted_data,
                 self.regularization_weighting:regularization_weighting}
```

Comment:
Similar comment to elsewhere in this review, but what is the interaction between need_ages and needing to adjust X by age?

Reply:
(I don't understand why one implies the other.)

Reply:
If it's just for the sparse correlation stuff, I'm a bit worried that:
1) Other models need age (e.g., the split-model multiplicative thing) but don't need age_adjusted data
2) Needing to handle age_adjusted_data is leaking all over our abstractions even though it's specific to one class

Reply:
one does not imply the other, i agree. perhaps there's a better way to refactor it?

Reply:
basically, models that require age_adjusted_X are a subset of models that need age.